### PR TITLE
[ENH] Improve issue templates

### PR DESCRIPTION
**Pull request by 0dminnimda**
I can see that some users still get confused and I think we can improve wording and some templates to be nicer to handle, so people could understand we what from them more easily and be less overwhelmed 

What I've changed:

- Removed blank issue and renamed "other" to be blank issue template with text
- Updated some titles, descriptions
- Separated environment section into `input` fields and made it so `[e.g. X]` are not there, but are just suggestions
- Renamed issue templates so it would be possible to sort them in any order we want (`name.yml` -> `{id}_name.yml`, github uses alphabetical ordering)

I temporarily made it a default branch so you can try those templates with [my fork](https://github.com/0dminnimda/cython/issues/new/choose)
Bug report will look like [this](https://github.com/0dminnimda/cython/issues/5#issue-1551318569)
image

image
image

> **Note** I don't think this vulnerability issue is gonna be in the cython repo, I tried to get rid of it, it stayed no matter what

image

### CODE REVIEW

**Comment by 0dminnimda** [comment]
I can understand that new bug report environment may not be welcomed, it can be excluded, the main point is to disable blank issue and make it into a template

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
-blank_issues_enabled: true
+blank_issues_enabled: false
 contact_links:
-  - name: Cython-users mailing list
+  - name: I NEED HELP
     url: https://groups.google.com/g/cython-users
-    about: Please ask and answer support questions here instead of on github.
+    about: Cython-users mailing list is a place to ask and answer support questions.
```
Comment (scoder) [inline .github/ISSUE_TEMPLATE/config.yml:5]:
```suggestion
    about: The Cython-users mailing list is a place to ask and answer support questions.
```

**Review by scoder — APPROVED** [review]
I think we should try this. Thanks.

File: .github/ISSUE_TEMPLATE/config.yml [hunk 2]
```diff
@@ -1,5 +1,5 @@
-blank_issues_enabled: true
+blank_issues_enabled: false
 contact_links:
-  - name: Cython-users mailing list
+  - name: I NEED HELP
```
[the remark this answers is not in the document]
Comment (0dminnimda) [reply .github/ISSUE_TEMPLATE/config.yml:3]: Let's try and see how it'll look

[the remark this answers is not in the document]
Comment (0dminnimda) [reply .github/ISSUE_TEMPLATE/config.yml]: image

[the remark this answers is not in the document]
Comment (0dminnimda) [reply .github/ISSUE_TEMPLATE/config.yml]: I don't hold string opinions about these names, if you are ok with it I am ok as well

**Comment by 0dminnimda** [comment]
Some feedback from other people probably can help here. Are there any more comments?

**Comment by da-woods** [comment]
I'm happy with these changes I think

**Comment by scoder** [comment]
Thanks

**Pull request merged**